MobileManager/Services/AppResourceService.cs

```diff
@@ -0,0 +1,154 @@
+        public AppResourceService(IRestClient restClient, IRepository<AppResourceInfo> appResourceRepository, IManagerLogger logger) : base(logger)
+        {
+            _logger = logger;
+            _restClient = restClient;
+            _appResourceRepository = appResourceRepository;
+            _storagePath = Path.Combine(Directory.GetCurrentDirectory(), "../_appFiles");
+        }
+
+        /// <inheritdoc />
+        public AppResourceInfo FindAppResource(string id)
+        {
+            return _appResourceRepository.Find(id);
+        }
+
+        /// <inheritdoc />
+        public bool DeleteAppResource(string id)
+        {
+            var appFolder = Path.Combine(_storagePath, id);
+            var result = _appResourceRepository.Remove(id);
+            Directory.Delete(appFolder, true);
+            return true;
+        }
+
+        /// <inheritdoc />
+        public async Task<AppResourceInfo> UploadAppResource(IFormFile file)
+        {
+            CreateStorageFolder();
+
+            var extension = Path.GetExtension(file.FileName).Trim('.').ToLower();
+            var hash = GetHashFromFile(file.OpenReadStream());
+            var folderPath = Path.Combine(_storagePath, hash);
+            var filePath = Path.Combine(folderPath, file.FileName);
+
+            if (!Directory.Exists(folderPath))
+            {
+                Directory.CreateDirectory(folderPath);
+                var newStream = new FileStream(filePath, FileMode.Create);
+                await file.CopyToAsync(newStream);
+                var appInfo = new AppResourceInfo(hash, file.FileName, filePath, extension, DateTime.Now);
+                _appResourceRepository.Add(appInfo);
+                newStream.Close();
+                return appInfo;
+            }
+
+            return FindAppResource(hash);
+        }
+
+        /// <inheritdoc />
+        public AppResourceInfo DownloadAppResourceFromUri(Uri uri)
+        {
+            /* 
+            * at first try HEAD HTTP request
+            * if is MD5 there we can use it without necessity to download whole file 
+            * It works for e.g. artifactory 
+            */
+            WebRequest request = WebRequest.Create(uri);
+            request.Method = "HEAD";
+            var md5 = request.GetResponse().Headers.Get("X-Checksum-Md5");
+            if (md5 != null)
+            {
+                var appResourceInfo = FindAppResource(md5);
+                if (appResourceInfo.Id == md5)
+                {
+                    return appResourceInfo;
+                }
+            }
+
+            var fileName = Path.GetFileName(uri.AbsolutePath);
+            var randomTemp = "_temp" + (new Random()).Next(10000);
+            var tempFolderPath = Path.Combine(_storagePath, randomTemp);
+            var tempFilePath = Path.Combine(tempFolderPath, fileName);
+
+            using (var client = new WebClient())
+            {
+                Directory.CreateDirectory(tempFolderPath);

```

Comment:
this should not be inside using as its not using the client

Reply:
will be fixed